internal/feature/features.go

```diff
@@ -52,6 +55,7 @@ type Feature = featuregate.Feature
 type Gate interface {
 	Enabled(Feature) bool
 	String() string
```

Comment:
🔧 I misunderstood what upstream's `String()` does. Let's explain it here.

```suggestion
	// String returns the features enabled or disabled by Set and SetFromMap.
	String() string
```

<p><br>
🤔 Maybe we drop String from the interface and pivot fully to context functions?
</p>

```go
// ShowAssigned returns the features enabled or disabled by Set and SetFromMap
// in the Gate contained in ctx.
func ShowAssigned(ctx context.Context) string

// ShowEnabled returns the features enabled in the Gate contained in ctx.
func ShowEnabled(ctx context.Context) string
```

Reply:
The only place we use `gate.String()` is in main.go when we report what the user set -- but if we're just telling the user what they set, why not just print the env var?

Reply:
Whatever the shape on the way in, we have a canonical representation in the logs.